src/main/java/com/dev/cinema/controller/OrderController.java

```diff
@@ -32,15 +33,17 @@ public OrderController(OrderService orderService, OrderMapper orderMapper,
     }
 
     @PostMapping("/complete")
-    public void create(@RequestParam Long userId) {
-        User user = userService.getById(userId);
+    public void create(Authentication authentication) {
+        String userEmail = ((UserDetails) authentication.getPrincipal()).getUsername();
```

Comment:
I moved this line into separate method as you repeat it below, but not critical at all...

Reply:
done